### ISSUE TITLE
Not fetching complete OffsetAndMetadata information

### ISSUE THREAD

**Issue by AnamikaN** [dpkp/kafka-python#2026]
when i run CLI command to describe consumer group, i get below data
GROUP                   TOPIC                                      PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG             CONSUMER-ID     HOST            CLIENT-ID
data-connect-dev3-group dev.dcSacramento.sds.resourceManager       4          2834            2861            27              -               -               -

But When i run 'list_consumer_group_offsets' mthod. it pulls below data for given consumer group.
TopicPartition(topic=u'dev.dcSacramento.sds.resourceManager', partition=5): OffsetAndMetadata(offset=2833, metadata=u'')

Basically with CLI command, i get additional details of LOG-END-OFFSET  LAG             CONSUMER-ID     HOST            CLIENT-ID. 

Any suggestions, how to fetch this info using api?

**Comment by jeffwidman** [comment]
Are you sure your consumer group is storing it's consumer offsets in Kafka? At first glance it appears you might be using the old consumer that stores its offsets in Zookeeper.

If still having issues, please include the broker version, and the `kafka-python` code that you're using as well as the CLI code in this case to make it more clear what specifically you're comparing (there are several CLI tools depending on what version of Kafka broker).

**Comment by AnamikaN** [comment]
We are using broker version 2.3.

`
 self.admin_client = KafkaAdminClient(bootstrap_servers=bootstrap_servers,
                                                 client_id=client_id,
                                                 security_protocol=security_protocol,
                                                 sasl_mechanism=sasl_mechanism,
                                                 sasl_plain_username=username,
                                                 sasl_plain_password=password,
                                                 ssl_check_hostname=False)`

`offset_info = self.admin_client.list_consumer_group_offsets(group_id="data-connect-dev3-group", group_coordinator_id=None, partitions=None)`

`print(offset_info)`

it prints -

{TopicPartition(topic=u'dev.dcSacramento.sds.resourceManager', partition=5): OffsetAndMetadata(offset=2833, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.transformationService', partition=3): OffsetAndMetadata(offset=468, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.resourceManager', partition=4): OffsetAndMetadata(offset=2834, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.transformationService', partition=0): OffsetAndMetadata(offset=458, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.transformationService', partition=1): OffsetAndMetadata(offset=466, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.transformationService', partition=6): OffsetAndMetadata(offset=468, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.transformationService', partition=7): OffsetAndMetadata(offset=466, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.resourceManager', partition=3): OffsetAndMetadata(offset=2835, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.transformationService', partition=4): OffsetAndMetadata(offset=460, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.resourceManager', partition=2): OffsetAndMetadata(offset=2830, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.transformationService', partition=5): OffsetAndMetadata(offset=455, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.resourceManager', partition=1): OffsetAndMetadata(offset=2827, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.resourceManager', partition=0): OffsetAndMetadata(offset=2828, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.resourceManager', partition=7): OffsetAndMetadata(offset=2824, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.resourceManager', partition=6): OffsetAndMetadata(offset=2832, metadata=u''), TopicPartition(topic=u'dev.dcSacramento.sds.transformationService', partition=2): OffsetAndMetadata(offset=470, metadata=u'')}

Below is the CLI command-

bin/kafka-consumer-groups --bootstrap-server 10.xxx.xxx.xxx:9093,10.xxx.xxx.xxx:9093,10.xxx.xxx.xxx:9093 -command-config config/consumer.properties --describe --group data-connect-dev3-group

it prints-

`GROUP                   TOPIC                                      PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG             CONSUMER-ID     HOST            CLIENT-ID
data-connect-dev3-group dev.dcSacramento.sds.resourceManager       4          2834            3903            1069            -               -               -
data-connect-dev3-group dev.dcSacramento.sds.transformationService 4          460             464             4               -               -               -
data-connect-dev3-group dev.dcSacramento.sds.resourceManager       2          2830            3917            1087            -               -               -
data-connect-dev3-group dev.dcSacramento.sds.resourceManager       1          2827            3902            1075            -               -               -
data-connect-dev3-group dev.dcSacramento.sds.transformationService 6          468             471             3               -               -               -
data-connect-dev3-group dev.dcSacramento.sds.transformationService 3          468             472             4               -               -               -
data-connect-dev3-group dev.dcSacramento.sds.transformationService 0          458             461             3               -               -               -
data-connect-dev3-group dev.dcSacramento.sds.resourceManager       5          2833            3917            1084            -               -               -
data-connect-dev3-group dev.dcSacramento.sds.resourceManager       0          2828            3911            1083            -               -               -
data-connect-dev3-group dev.dcSacramento.sds.transformationService 5          455             458             3               -               -               -
data-connect-dev3-group dev.dcSacramento.sds.resourceManager       7          2824            3898            1074            -               -               -
data-connect-dev3-group dev.dcSacramento.sds.transformationService 1          466             469             3               -               -               -
data-connect-dev3-group dev.dcSacramento.sds.resourceManager       3          2835            3919            1084            -               -               -
data-connect-dev3-group dev.dcSacramento.sds.transformationService 7          466             469             3               -               -               -
data-connect-dev3-group dev.dcSacramento.sds.transformationService 2          470             473             3               -               -               -
data-connect-dev3-group dev.dcSacramento.sds.resourceManager       6          2832            3915            1083            -               -               -`

My question is, how to get LOG-END-OFFSET  and LAG using api?

I found a workaround for this using below code. Below code is creating consumer, subscribing to topic, then pulling end offsets for all partition. The cons of below process is that we have to run below code for all topics and all partitions for that consumer group, then calculate lag for it.

`consumer = KafkaConsumer(bootstrap_servers=bootstrap_servers,
                                     client_id=client_id,
                                     security_protocol=security_protocol,
                                     sasl_mechanism=sasl_mechanism,
                                     sasl_plain_username=username,
                                     sasl_plain_password=password,
                                     group_id="consumer_group_id",
                                     ssl_check_hostname=False)`
 `topic = "dev.dcSacramento.sds.resourceManager"`
`consumer.subscribe("dev.dcSacramento.sds.resourceManager")`
 `partitions = [TopicPartition(topic, 0), TopicPartition(topic, 1), TopicPartition(topic, 2),
                          TopicPartition(topic, 3), TopicPartition(topic, 4), TopicPartition(topic, 5),
                          TopicPartition(topic, 6), TopicPartition(topic, 7)]`
`print(consumer.end_offsets(partitions))`

My question is, how to get LOG-END-OFFSET  and LAG using api?

**Comment by jeffwidman** [comment]
The broker highwater offset is fetched using a different API, and then you have to manually calculate the lag by doing the subtraction. That's all the CLI tool is doing under the covers, it's also using multiple broker API calls.

Here's an example script that fetches consumer offset, broker highwater offset, and calculates consumer lag... this also does it in parallel using callbacks/futures so that it scales quite well: 
https://github.com/DataDog/integrations-core/blob/567ada043165f03704a63908b3439bd83765d238/kafka_consumer/datadog_checks/kafka_consumer/kafka_consumer.py

If you look through this, you'll see the various APIs that give you what you want.
